tests/integration/helpers.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from pathlib import Path
+from urllib.parse import urljoin
+
+import requests
+import yaml
+
+logger = logging.getLogger(__name__)
+
+
+async def get_unit_address(ops_test, app_name: str, unit_num: int) -> str:
+    status = await ops_test.model.get_status()  # noqa: F821
+    return status["applications"][app_name]["units"][f"{app_name}/{unit_num}"]["address"]
+
+
+async def mimir_endpoint_request(ops_test, app_name: str, endpoint: str, unit_num: int = 0):
```

Comment:
Have you considered a client class?
- [async ex.](https://github.com/canonical/prometheus-k8s-operator/blob/main/tests/integration/workload.py)
- [sync ex.](https://github.com/canonical/prometheus-k8s-operator/blob/main/src/prometheus_server.py)